blccpsapibukkit/pom.xml

```diff
@@ -35,14 +35,14 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>1.7</source>
+                    <target>1.7</target>
```

Comment:
I just tested it myself on a clean install, I can build it fine with Java 6.

This was kept in order to make sure this plugin is compatible with most setups, as some servers may still be using Java 6.

Can you provide us the complete Maven output ?

Reply:
The `maven-compiler-plugin` uses `javac` to compile the project. There's an OpenJDK policy outlined in [JEP 182: Policy for Retiring javac -source and -target Options](https://openjdk.java.net/jeps/182) for retiring old `javac` `-source` and `-target` options. Support for a source or target of `1.5` was removed in JDK 9 and to comply with this policy, [support for a source or target of `1.6` was already removed in JDK 12](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8028563). 

I use JDK 13 and JDK 14 will become generally available in the first quarter of 2020. 

[According to MCStats.org only 0.01% (10 of over 10,000 tracked servers) still use Java 6. Even Java 7 is barely used.](http://mcstats.org/global/)
![grafik](https://user-images.githubusercontent.com/12609161/69224498-f9062080-0b7c-11ea-8b75-7bd5d2d403a7.png)

[bStats.org doesn't even track any servers that are still running on Java 6 or 7.](https://bstats.org/global/bukkit)

And by the way, the `source` and `target` options of the BungeeCord plugin are also set to `1.7`.

For these reasons I'd strongly suggest to update to Java 7 so that third party developers (like me) can also contribute to this project.

```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for net.badlion:blcmodapivelocity:jar:1.2.0
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 24, column 21
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for net.badlion:blcmodapibungee:jar:1.2.0
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 34, column 21
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for net.badlion:blcmodapibukkit:jar:1.2.0
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 34, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] badlionclientmodapi                                                [pom]
[INFO] blcmodapivelocity                                                  [jar]
[INFO] blcmodapibungee                                                    [jar]
[INFO] blcmodapibukkit                                                    [jar]
[INFO] 
[INFO] ------------------< net.badlion:badlionclientmodapi >-------------------
[INFO] Building badlionclientmodapi 1.2.0                                 [1/4]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ badlionclientmodapi ---
[INFO] 
[INFO] -------------------< net.badlion:blcmodapivelocity >--------------------
[INFO] Building blcmodapivelocity 1.2.0                                   [2/4]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from : https://repo.velocitypowered.com/snapshots/com/velocitypowered/velocity-api/1.0.4-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://repo.velocitypowered.com/snapshots/com/velocitypowered/velocity-api/1.0.4-SNAPSHOT/maven-metadata.xml (1.2 kB at 1.7 kB/s)
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ blcmodapivelocity ---
[INFO] Deleting C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapivelocity\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ blcmodapivelocity ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapivelocity\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ blcmodapivelocity ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapivelocity\target\classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ blcmodapivelocity ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapivelocity\src\test\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ blcmodapivelocity ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ blcmodapivelocity ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ blcmodapivelocity ---
[INFO] Building jar: C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapivelocity\target\badlionclientmodapivelocity.jar
[INFO] 
[INFO] --------------------< net.badlion:blcmodapibungee >---------------------
[INFO] Building blcmodapibungee 1.2.0                                     [3/4]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-api/1.12-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-api/1.12-SNAPSHOT/maven-metadata.xml (1.2 kB at 1.1 kB/s)
[INFO] Downloading from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-parent/1.12-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-parent/1.12-SNAPSHOT/maven-metadata.xml (602 B at 2.1 kB/s)
[INFO] Downloading from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-chat/1.12-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-chat/1.12-SNAPSHOT/maven-metadata.xml (1.2 kB at 4.5 kB/s)
[INFO] Downloading from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-config/1.12-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-config/1.12-SNAPSHOT/maven-metadata.xml (1.2 kB at 4.6 kB/s)
[INFO] Downloading from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-event/1.12-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-event/1.12-SNAPSHOT/maven-metadata.xml (1.2 kB at 4.4 kB/s)
[INFO] Downloading from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-protocol/1.12-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://oss.sonatype.org/content/repositories/snapshots/net/md-5/bungeecord-protocol/1.12-SNAPSHOT/maven-metadata.xml (1.2 kB at 4.5 kB/s)
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ blcmodapibungee ---
[INFO] Deleting C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapibungee\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ blcmodapibungee ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource to .
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ blcmodapibungee ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapibungee\target\classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ blcmodapibungee ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapibungee\src\test\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ blcmodapibungee ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ blcmodapibungee ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ blcmodapibungee ---
[INFO] Building jar: C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapibungee\target\badlionclientmodapibungee.jar
[INFO] 
[INFO] --------------------< net.badlion:blcmodapibukkit >---------------------
[INFO] Building blcmodapibukkit 1.2.0                                     [4/4]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from : https://hub.spigotmc.org/nexus/content/repositories/snapshots/org/bukkit/bukkit/1.12.2-R0.1-SNAPSHOT/maven-metadata.xml
[INFO] Downloaded from : https://hub.spigotmc.org/nexus/content/repositories/snapshots/org/bukkit/bukkit/1.12.2-R0.1-SNAPSHOT/maven-metadata.xml (1.5 kB at 2.2 kB/s)
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ blcmodapibukkit ---
[INFO] Deleting C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapibukkit\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ blcmodapibukkit ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource to .
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ blcmodapibukkit ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientmodapi\blcmodapibukkit\target\classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for badlionclientmodapi 1.2.0:
[INFO] 
[INFO] badlionclientmodapi ................................ SUCCESS [  0.135 s]
[INFO] blcmodapivelocity .................................. SUCCESS [  3.775 s]
[INFO] blcmodapibungee .................................... SUCCESS [  3.170 s]
[INFO] blcmodapibukkit .................................... FAILURE [  0.886 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.087 s
[INFO] Finished at: 2019-11-20T10:16:42+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project blcmodapibukkit: Compilation failure: Compilation failure: 
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :blcmodapibukkit
```

Reply:
I fixed the warnings from the beginning of the build log in a new commit (see below).